src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -777,9 +792,8 @@ def _extractValuesFromJobParams(self, params):
         submissionDict["jobID"] = params.get("JobID")
         if not submissionDict["jobID"]:
             msg = "Job has not JobID defined in JDL parameters"
-            self.jobReport.setJobStatus(status=JobStatus.FAILED, minorStatus=msg)
```

Comment:
Where did this go?

Reply:
This is here now: https://github.com/DIRACGrid/DIRAC/pull/7415/files#diff-81052c54bdd87d4f640519488aff350cadab94a15eb0aadd9d30166ab9d23dabR253

Do you prefer to have it within the method?
First, I tried to put these `setJobStatus` calls in the `execute` method to get a "clear" overview of them. But then I realized it was not easy to do it everywhere (e.g. `checkMatcherInfo()`).

Reply:
No, it's fine!